File: examples/javascript/get_tasks.js
const BASE_URL = "https://wattcoin-production-81a7.up.railway.app";

async function getTasks() {
  try {
    const response = await fetch(`${BASE_URL}/tasks`);
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }
    const data = await response.json();
    console.log(data);
  } catch (error) {
    console.error("Error fetching tasks:", error);
  }
}

getTasks();

File: examples/python/get_tasks.py
import requests

BASE_URL = "https://wattcoin-production-81a7.up.railway.app"

def get_tasks():
    """Fetch all available tasks."""
    response = requests.get(f"{BASE_URL}/tasks")
    response.raise_for_status()
    return response.json()

if __name__ == "__main__":
    tasks = get_tasks()
    print(tasks)
